src/llmq/chainlocks.cpp

```diff
@@ -620,11 +620,8 @@ void CChainLocksHandler::Cleanup()
         return;
     }
 
-    {
-        LOCK(cs);
-        if (GetTimeMillis() - lastCleanupTime < CLEANUP_INTERVAL) {
-            return;
-        }
+    if (GetTimeMillis() - lastCleanupTime < CLEANUP_INTERVAL) {
+        return;
```

Comment:
since we make `lastCleanupTime` atomic, we can re-order locks below such as:
```
LOCK(cs);
LOCK2(cs_main, mempool.cs);
```
so, in this case we won't lock cs_main and mempool.cs if `cs` is locked.

thoughts?

Reply:
can you explain further *why* we can reorder those now? I'm not seeing it

Reply:
nevermind, it can not be re-ordered:
```
Previous lock order was:
(2) 'cs_main' in miner.cpp:130 (in thread 'httpworker.1')
'm_mempool.cs' in miner.cpp:130 (in thread 'httpworker.1')
(1) 'cs' in llmq/chainlocks.cpp:588 (in thread 'httpworker.1')
Current lock order is:
(1) 'cs' in llmq/chainlocks.cpp:628 (in thread 'cl-schdlr')
(2) 'cs_main' in llmq/chainlocks.cpp:629 (in thread 'cl-schdlr')
```